### DISCUSSION TITLE
If I delete data from admin dashboard or from database, in frontend I am not seeing updated data.

### DISCUSSION THREAD

**Discussion by cabbageofficial** [urql-graphql/urql#3066] [Q&A]
Hello, I am facing one new problem with `urql`. When I delete any data from another dashboard or from database, the updated data not been updated into urql query. But from getServerSideProps it fetch new data.

Here is the video-

[Urql-data-from-postgree.webm](https://user-images.githubusercontent.com/88053948/225683038-e8a93af3-5b7a-4f79-9cf1-3f535d209e2b.webm)

Here is my code-

export const getServerSideProps: GetServerSideProps = async (ctx) => {
    const { data } = await client.query(
        GET_STUDENT_LIST,
        { studentPrams: { name: "", studentId: "", class: "" } }, {
        fetchOptions: {
            headers: {
                cookie: ctx.req.headers.cookie as string,
                "user-agent": ctx.req.headers["user-agent"] as string,
                "x-forwarded-for": ctx.req.socket.remoteAddress as string
            },
        },
        requestPolicy: "network-only",
    }).toPromise();
    console.log(data);
    return { props: { urqlState: ssrCache?.extractData() } }
}

`GET_STUDENT_QUERY`-

```
export const GET_STUDENT_LIST = gql`
query getStudents($studentPrams: StudentPrams!) {
  getStudents(studentPrams: $studentPrams) {
    id
    studentId
    name
    fatherName
    dob
    number
    createdBy {
      name
      phone
    }
    __typename
  }
}
`;
```

Here __typename is `Student`-

And one tree label component undet above page-

` const [{ data, error, fetching }, refetch] = useQuery<GetStudentListData>({ query: GET_STUDENT_LIST, variables: { studentPrams: { name: getValues("name") ?? "", studentId: getValues("studentId") ?? "", class: getValues("class") ?? "" } } });`

And here is my `client.ts`-

```
import { createClient, ssrExchange, dedupExchange, cacheExchange, fetchExchange, errorExchange } from "urql";

const isServerSide = typeof window === "undefined";

const ssrCache = ssrExchange({ isClient: !isServerSide });

const client = createClient({
    url: "http://localhost:3001/ecampus",
    exchanges: [
        dedupExchange,
        errorExchange({
            onError: (error) => {
                error.message = error.message.replace(/^\[.*\]\s*/, "");
            }
        }),
        ssrCache,
        cacheExchange,
        fetchExchange
    ],
    fetchOptions: {
        credentials: "include"
    },
});

export { client, ssrCache };
```

I also use `next-urql`, I am facing same issues.

Here my question is, why `useQuery` not getting new data, where `getServerSideProps` fetching the new data.

**Comment by JoviDeCroock** [comment]
You did not post your mutation again but probably same issue as last time 😅 you might return a scalar/... rather than the type we can infer to invalidate your cache

**Reply by cabbageofficial** [reply]
Nope, I can successfully solve previous problem. When I running my mutation it's working fine. But When I delete it from another dashboard or directly from database, Here I am not running any mutation. But data not get updated. 

If I delete from this project by click delete button, here it running a mutation for deleting the project. In this time, I am getting updated data. But If I delete it from database, it not showing updated data, but it should!

Here I have two dashboard, one main dashboard another is super dashboard. If an admin delete one student from super dashboard, it will be update from that super dashboard only, but in the main dashboard it will not be updated. I thing I can clear to you this matter.

Believe me, I can solve previous problem. It is not previous problem, it is new. Here just give me some idea.

**Reply by kitten** [reply]
> ere I have two dashboard, one main dashboard another is super dashboard. If an admin delete one student from super dashboard, it will be update from that super dashboard only, but in the main dashboard it will not be updated.

What do you mean by that?
How would it update from one dashboard to another?
You mean you have two tabs open and you change something in one?

If that's the case then the other tab won't know that something has changed. We currently don't have any cross-tab functionality.

Or do you mean something else?

**Reply by JoviDeCroock** [reply]
I mean, that's correct right, this is entirely the same issue as the previous thing... You are relying on `getServerSideProps` for an action that can't evict the cached entity in the `cacheExchange`. This is one of the reasons why I personally advise against relying on `getServerSideProps` for most applications 😅 In GraphQL there is a notion of `subscriptions` which can help you monitor what outside actors are doing to the data you are controlling, the `cache` you have on the client is relying on being aware of these interactions.

What I catch here is that you might prefer removing the `cacheExchange` on the client _or_ using the `requestPolicyExchange` to upgrade your request-policy to `cache-and-network` whenever appropriate

**Reply by cabbageofficial** [reply]
Kitten, I don't means that, but if another tab, I am changing my route, changing route means, I am going another page, and come back to the previous page, as I using srr data fetching, then it should updated new data, please see my video.

**Reply by cabbageofficial** [reply]
I understand JoviDeCroock advise, I should not use ssr data fetching, or I have to use cache-and-network as requestpolicy.
